File: Web App/SisMedCare/common/models/Paciente.php
<?php

namespace common\models;

use Yii;

/**
 * This is the model class for table "pacientes".
 *
 * @property int $id
 * @property int $user_id
 * @property string $nome_completo
 * @property string $data_nascimento
 * @property string $sexo
 * @property string $numero_utente
 * @property string $telemovel
 * @property string $morada
 * @property float|null $altura
 * @property float|null $peso
 * @property string|null $alergias
 * @property string|null $doencas_cronicas
 * @property string $data_registo
 *
 * @property Consulta[] $consulta
 * @property Prescricao[] $prescricao
 * @property RegistoToma[] $registoToma
 * @property User $user
 */
class Paciente extends \yii\db\ActiveRecord
{


    /**
     * {@inheritdoc}
     */
    public static function tableName()
    {
        return 'pacientes';
    }

    /**
     * {@inheritdoc}
     */
    public function rules()
    {
        return [
            [['user_id', 'nome_completo', 'data_nascimento', 'sexo', 'numero_utente', 'telemovel', 'morada'], 'required'],
            [['altura', 'peso', 'alergias', 'doencas_cronicas'], 'default', 'value' => null],
            [['user_id', 'numero_utente'], 'integer'],
            [['altura', 'peso'], 'number'],
            [['data_nascimento', 'data_registo'], 'safe'],
            [['nome_completo', 'morada'], 'string', 'max' => 255],
            [['sexo'], 'string', 'max' => 1],
            [['telemovel'], 'string', 'max' => 20],
            [['alergias', 'doencas_cronicas'], 'string', 'max' => 255],

            [['numero_utente'], 'unique', 'message' => 'Este número de utente já está registado.'],
            [['telemovel'], 'unique', 'message' => 'Este número de telemóvel já está registado.'],

            [['user_id'], 'exist', 'skipOnError' => true, 'targetClass' => User::class, 'targetAttribute' => ['user_id' => 'id']],
        ];
    }

    /**
     * {@inheritdoc}
     */
    public function attributeLabels()
    {
        return [
            'id' => 'ID Paciente',
            'user_id' => 'ID Utilizador',
            'nome_completo' => 'Nome Completo',
            'data_nascimento' => 'Data de Nascimento',
            'sexo' => 'Sexo',
            'numero_utente' => 'Número de Utente',
            'telemovel' => 'Telemóvel',
            'morada' => 'Morada',
            'altura' => 'Altura',
            'peso' => 'Peso',
            'alergias' => 'Alergias',
            'doencas_cronicas' => 'Doenças Crónicas',
            'data_registo' => 'Data de Registo',
        ];
    }

    /**
     * Gets query for [[Consultas]].
     *
     * @return \yii\db\ActiveQuery
     */
    public function getConsultas()
    {
        return $this->hasMany(Consulta::class, ['paciente_id' => 'id']);
    }

    /**
     * Gets query for [[Prescricoes]].
     *
     * @return \yii\db\ActiveQuery
     */
    public function getPrescricoes()
    {
        return $this->hasMany(Prescricao::class, ['paciente_id' => 'id']);
    }

    /**
     * Gets query for [[RegistosTomas]].
     *
     * @return \yii\db\ActiveQuery
     */
    public function getRegistosTomas()
    {
        return $this->hasMany(RegistoToma::class, ['paciente_id' => 'id']);
    }

    /**
     * Gets query for [[User]].
     *
     * @return \yii\db\ActiveQuery
     */
    public function getUser()
    {
        return $this->hasOne(User::class, ['id' => 'user_id']);
    }
}

File: Mobile App/app/src/main/java/pt/ipleiria/estg/dei/sismedcare/modelo/Paciente.java
package pt.ipleiria.estg.dei.sismedcare.modelo;

public class Paciente {
    private int id;
    private int userId;
    private String username;
    private String email;
    private String nomeCompleto;
    private String dataNascimento; // YYYY-MM-DD
    private String sexo; // "M", "F", "O"
    private String numeroUtente;
    private String telemovel;
    private String morada;
    private float altura;
    private float peso;
    private String alergias;
    private String doencasCronicas;
    private String dataRegisto;

    private User user; // referência ao User associado

    public Paciente(int id, int userId, String nomeCompleto, String dataNascimento, String sexo,
                    String numeroUtente, String telemovel, String morada, float altura, float peso,
                    String alergias, String doencasCronicas, String dataRegisto, User user) {
        this.id = id;
        this.userId = userId;
        this.nomeCompleto = nomeCompleto;
        this.dataNascimento = dataNascimento;
        this.sexo = sexo;
        this.numeroUtente = numeroUtente;
        this.telemovel = telemovel;
        this.morada = morada;
        this.altura = altura;
        this.peso = peso;
        this.alergias = alergias;
        this.doencasCronicas = doencasCronicas;
        this.dataRegisto = dataRegisto;
        this.user = user;
    }

    public int getId() { return id; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public String getNomeCompleto() { return nomeCompleto; }
    public String getDataNascimento() { return dataNascimento; }
    public String getSexo() { return sexo; }
    public String getNumeroUtente() { return numeroUtente; }
    public String getTelemovel() { return telemovel; }
    public String getMorada() { return morada; }

    public void setId(int id) { this.id = id; }
}
